corehq/apps/userreports/views.py

```diff
@@ -1218,6 +1218,8 @@ def post(self, request, *args, **kwargs):
                     return HttpResponseRedirect(reverse(
                         EditDataSourceView.urlname, args=[self.domain, config._id])
                     )
+            else:
+                messages.error(request, _('Data source not saved. Please check the form for errors.'))
```

Comment:
Which form is this?

Reply:
The form is visible on the page, when you are editing a data source.